lib/src/supabase_auth.dart

```diff
@@ -303,12 +318,92 @@ extension GoTrueClientSignInProvider on GoTrueClient {
       scopes: scopes,
       queryParams: queryParams,
     );
-    final url = Uri.parse(res.url!);
-    final result = await launchUrl(
-      url,
-      mode: authScreenLaunchMode,
-      webOnlyWindowName: '_self',
+    final uri = Uri.parse(res.url!);
+
+    final willOpenWebview = authScreenLaunchMode == LaunchMode.inAppWebView &&
+        context != null &&
+        !kIsWeb && // `Platform.isIOS` throws on web, so adding a guard for web here.
+        Platform.isIOS;
+
+    if (willOpenWebview) {
+      Navigator.of(context).push(PageRouteBuilder(
+          pageBuilder: (context, animation, secondaryAnimation) {
+        return _OAuthSignInWebView(oAuthUri: uri, redirectTo: redirectTo);
+      }));
+      return true;
```

Comment:
Opening the OAuth screen in full screen instead of bottom sheet so that Cupertino app users can take advantage of this nice feature as well!